compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -906,7 +906,7 @@ where
                         if i == tag_field {
                             return TyMaybeWithLayout::TyAndLayout(tag_layout(tag));
                         }
-                        TyMaybeWithLayout::Ty(args.as_generator().prefix_tys().nth(i).unwrap())
+                        TyMaybeWithLayout::Ty(*args.as_generator().prefix_tys().get(i).unwrap())
```

Comment:
```suggestion
                        TyMaybeWithLayout::Ty(args.as_generator().prefix_tys()[i])
```

Reply:
ty::Lists are not indexable

Reply:
huh. I thought their `Deref<Target = [T]>` impl would make that "just work"

Reply:
ok i lied, for some reason it works now